script.js

```diff
@@ -1,4 +1,130 @@
+      // Player 1 mode 1 -> mode 2
+      currentGameMode = "playerOneMode2";
+    }
+    // Loop 2
+    // Player 1 mode 2: Choose which dice as first
+  } else if (currentGameMode == "playerOneMode2") {
+    // if (input != "1" || input != "2") {
+    //   outputValue = `Please choose the order of the dice by entering "1" or "2" ONLY.`;
+    // }
+    // Player 1 choose 1st dice
+    if (input == "1") {
+      playerOneFinalDice = `${playerOneFirstDice}${playerOneSecondDice}`;
+      console.log(playerOneFinalDice);
+      outputValue = `Player 1, you chose Dice ${input} first. <br><br> Your number is ${playerOneFinalDice}. <br><br> It is now Player 2's turn. <br><br> Please press "Submit" to continue!`;
+      // Player 1 choose 2nd dice
+    } else if (input == "2") {
+      playerOneFinalDice = `${playerOneSecondDice}${playerOneFirstDice}`;
+      console.log(playerOneFinalDice);
+      outputValue = `Player 1, you chose Dice ${input} first. <br><br> Your number is ${playerOneFinalDice}. <br><br> It is now Player 2's turn. <br><br> Please press "Submit" to continue!`;
+    } else {
+      return `Please input 1 or 2 only.`;
+    }
+
+    currentGameMode = "playerTwoMode1";
+    // Loop 3
+    // Player 1 mode ended. Start Player 2 mode 1
+  } else if (currentGameMode == "playerTwoMode1") {
+    // User cannot key in anything except pressing submit button
+    if (input != "") {
+      outputValue = "Please roll the dice by clicking 'Submit'.";
+    } else {
+      // Player 2 mode 1: Roll 2 dices
+      playerTwoFirstDice = rollDice1();
+      playerTwoSecondDice = rollDice2();
+      console.log(playerTwoFirstDice);
+      console.log(playerTwoSecondDice);
+      outputValue = `Welcome Player 2. <br><br> You rolled ${playerTwoFirstDice} for Dice 1 and ${playerTwoSecondDice} for Dice 2. <br><br> Choose the order of the dice by entering "1" or "2" ONLY.`;
+      // Player 2 mode 1 -> mode 2
+      currentGameMode = "playerTwoMode2";
+    }
+    // Loop 4
+    // Player 2 mode 2: Choose which dice as first
+  } else if (currentGameMode == "playerTwoMode2") {
+    // Player 2 choose 1st dice
+    if (input == "1") {
+      playerTwoFinalDice = `${playerTwoFirstDice}${playerTwoSecondDice}`;
+      console.log(playerTwoFinalDice);
+      outputValue = `Player 2, you chose Dice ${input} first. <br><br> Your number is ${playerTwoFinalDice}. <br><br> Press "Submit" to see who win!`;
+      // Player 1 choose 2nd dice
+    } else if (input == "2") {
+      playerTwoFinalDice = `${playerTwoSecondDice}${playerTwoFirstDice}`;
+      console.log(playerTwoFinalDice);
+      outputValue = `Player 2, you chose Dice ${input} first. <br><br> Your number is ${playerTwoFinalDice}. <br><br> Press "Submit" to see who win!`;
+    } else {
+      return `Please input 1 or 2 only.`;
```

Comment:
Since this 2 big loops are similar, you can abstract them into functions to avoid repeating code